auto_dev/test.py

```diff
@@ -14,6 +14,11 @@
 """
 
 
+def get_test_cpu_count() -> str:
+    """Determine how many CPUs to use for pytest-xdist."""
+    return str(cpu_count()) if os.getenv("CI") else "auto"
```

Comment:
on local this ensures not all cores will be used (3/4th I believe), while on the CI we take all that are available